### ISSUE TITLE
A deny under the all-tools `*` category escapes the shadowed-rule warning

### ISSUE THREAD

**Issue by dyoshikawa** [dyoshikawa/rulesync#2825]
A canonical `deny` written under the all-tools `*` category is reported by the deepagents and Warp permissions adapters only as "cannot be represented and were skipped", never as the stronger "those commands run without a prompt" warning that a `bash`-category deny gets when an `allow` beside it shadows it.

Repro with `.rulesync/permissions.jsonc`:

```jsonc
{
  "permission": {
    "*": { "rm *": "deny" },
    "bash": { "rm *": "allow" }
  }
}
```

`rulesync generate --targets deepagents --global` writes:

```toml
[shell]
allow_list = [ "rm" ]
```

and warns only that the `*` deny rules could not be represented. But the executable-name reduction has inverted the author's intent exactly as it would for a `bash`-category deny: `rm` is now auto-approved with no prompt. The narrow-vs-broad collision warning exists for precisely this case and does not fire.

`src/features/permissions/deepagents-permissions.ts` folds every non-`bash` category into `hasForeignDeny`, and `src/features/permissions/warp-permissions.ts` treats `*` the same way, so this is a shared convention rather than a bug in one adapter. Deciding it centrally — should the `*` category's rules be considered alongside the tool's own category when checking for a shadowed rule? — is the useful fix, and it needs the same answer in both adapters.

Note that the `all` sentinel is already blocked correctly by `hasForeignDeny`; only the per-command shadow warning is missing.

Found while reviewing #2824.
